pkg/backend/apply.go

```diff
@@ -76,10 +76,6 @@ func PreviewThenPrompt(ctx context.Context, kind apitype.UpdateKind, stack Stack
 	// create a channel to hear about the update events from the engine. this will be used so that
 	// we can build up the diff display in case the user asks to see the details of the diff
 	eventsChannel := make(chan engine.Event)
-	defer func() {
```

Comment:
Can we add a small comment to explain that we explicitly don't defer the close here and we have to manually close on exits, because it is not safe to close when panicing?